components/automate-cli/cmd/chef-automate/cleanup.go

```diff
@@ -0,0 +1,172 @@
+}{}
+
+type SshClient struct {
+	sshUser string
+	sshPort string
+	sshKeyFile string
+}
+
+func init() {
+	RootCmd.AddCommand(cleanupCmd)
+	cleanupCmd.PersistentFlags().BoolVar(&cleanupFlags.onprem, "onprem", false, "Cleaning up all the instances related to onprem ")
+
+}
+
+var cleanupCmd = &cobra.Command {
+		Use:  "cleanup",
+	    Short: "cleanup the Automate HA instances",
+	    Long:  "cleaning up the instance of all the Automate HA related Applications.",
+		Annotations: map[string]string{
+			NoCheckVersionAnnotation: NoCheckVersionAnnotation,
+		},
+		RunE: runCleanupCmd,
+		Hidden: true,
+		
+}
+
+const (
+		FRONTENDCLEANUP_COMMANDS = `
+		sudo systemctl stop chef-automate;
+		sudo rm -rf /hab;
+		`
+	
+		BACKENDCLEANUP_COMMANDS = `
+		sudo systemctl stop hab-sup;
+		sudo rm -rf /hab;   
+		`
+)
+
+func runCleanupCmd(cmd *cobra.Command, args []string) error {
+	infra, err := getAutomateHAInfraDetails()
+	if err != nil {
+		return err
+	}
+	if infra != nil {
+		sshInfo :=SshClient{
+			sshUser: infra.Outputs.SSHUser.Value,
+			sshPort: infra.Outputs.SSHPort.Value,
+			sshKeyFile: infra.Outputs.SSHKeyFile.Value,
+		}
+
+			writer.Printf(strings.Join(args, ""))
+		if isA2HARBFileExist() {
+			if cleanupFlags.onprem {
+				automateIps := infra.Outputs.AutomatePrivateIps.Value
+				chefserverIps := infra.Outputs.AutomatePrivateIps.Value
+				postgresqlIps := infra.Outputs.PostgresqlPrivateIps.Value
+				opensearchIps := infra.Outputs.OpensearchPrivateIps.Value
+				for i := 0; i < len(automateIps); i++ {
+					servername := "Automate"
+					err := executeCleanupOnRemote(sshInfo, automateIps[i], servername, FRONTENDCLEANUP_COMMANDS)
+					if err !=nil {
+						writer.Errorf("%s", err)
+						return err
+					}
+				}
+				for i := 0; i < len(chefserverIps); i++ {
+					servername := "chef server"
+					err := executeCleanupOnRemote(sshInfo, chefserverIps[i], servername, FRONTENDCLEANUP_COMMANDS)
+					if err !=nil {
+						writer.Errorf("%s", err)
+						return err
+					}
+				}
+				for i := 0; i < len(postgresqlIps); i++ {
+					servername := "postgresql"
+					err := executeCleanupOnRemote(sshInfo, postgresqlIps[i], servername, BACKENDCLEANUP_COMMANDS)
+					if err !=nil {
+						writer.Errorf("%s", err)
+						return err
+					}
+				}
+				for i := 0; i < len(opensearchIps); i++ {
+					servername := "opensearch"
+					err := executeCleanupOnRemote(sshInfo, opensearchIps[i], servername, BACKENDCLEANUP_COMMANDS)
+					if err !=nil {
+						writer.Errorf("%s", err)
+						return err
+					}
+				}
+				cleanUpScript := "hab pkg uninstall chef/automate-ha-deployment"
+				args := []string{
+					"-c",
+					cleanUpScript,
+				}
+				err := executeCommand("/bin/sh", args, "")
+				if err != nil {
+					return err
+				}
+				return err
+			}
+		}
+	} else {
+		fmt.Println("\nCleanup cannot be done")
+	}
+	return err
+}
+
+func executeCleanupOnRemote(sshDetails SshClient, ip string, servername string, commands string) error {
```

Comment:
If #7470 is merged before this PR, use 'ConnectAndExecuteCommandOnRemote' from config.go. All params are generalised there

Reply:
okay